inc/php_basic_auth/namespace.php

```diff
@@ -0,0 +1,104 @@
+<?php
+
+namespace Altis\Security\PHP_Basic_Auth;
+
+use function Altis\get_config;
+use function Altis\get_environment_type;
+
+function bootstrap() {
+	add_action( 'hmauth_action_before_dev_env_check', __NAMESPACE__ . '\\check_environment_credentials' );
+}
+
+/**
+ * Conditionally short-circuit the basic authentication settings based on the
+ * composer.json configuration.
+ */
+function check_environment_credentials() {
+	// Collect the environment overrides.
+	$env_local = get_config()['environments']['local']['modules']['security']['php-basic-auth'];
```

Comment:
This isn't going to be set necessarily, depending on the environment type the `composer.json` is merged recursively over the defaults so you can access the derived setting via `get_config()['modules']['security']['php-basic-auth']`.

Reply:
Looking at it I don't think you need this function, just the `define_constants()` one will do.

For reference this is what happens when you call `Altis\get_config()` for the first time:

https://github.com/humanmade/altis-core/blob/master/inc/namespace.php#L62-L64